handler/src/test/java/io/netty/handler/ssl/SniHandlerTest.java

```diff
@@ -229,4 +250,112 @@ protected void configurePipeline(ChannelHandlerContext ctx, String protocol) thr
             group.shutdownGracefully(0, 0, TimeUnit.MICROSECONDS);
         }
     }
+
+    @Test
+    public void testReplaceHandler() throws Exception {
+        final String sniHost = "sni.netty.io";
+        LocalAddress address = new LocalAddress("test");
+        EventLoopGroup group = new DefaultEventLoopGroup(1);
+        Channel sc = null;
+        Channel cc = null;
+
+        SelfSignedCertificate cert = new SelfSignedCertificate();
+
+        try {
+            final SslContext sslServerContext = SslContextBuilder
+                    .forServer(cert.key(), cert.cert())
+                    .sslProvider(SslProvider.OPENSSL)
```

Comment:
before this test runs you need to run an `assumeTrue(OpenSsl.isAvailable());`

consider extending this class an adding opnessl specific tests (following the same pattern as `OpenSslEngineTest`)